node/node_test.go

```diff
@@ -244,6 +245,61 @@ func TestRegisterProtocols_OpenRPC_WS(t *testing.T) {
 	}
 }
 
+func TestDiscoverCompliesOpenRPCSpec(t *testing.T) {
+	stack, err := New(testNodeConfig())
+	if err != nil {
+		t.Fatalf("failed to create protocol stack: %v", err)
+	}
+	defer stack.Close()
+
+	datadir := filepath.Join(os.TempDir(), "node-test")
+	defer os.RemoveAll(datadir)
+
+	stack.config = &DefaultConfig
+	stack.config.HTTPHost = DefaultHTTPHost
+	// enable all available APIs
+	stack.config.HTTPModules = []string{"admin", "debug", "eth", "ethash", "miner", "net", "personal", "rpc", "trace", "txpool", "web3"}
+	stack.config.DataDir = datadir
+
+	if err := stack.Start(); err != nil {
+		t.Fatal(err)
+	}
+	defer stack.Close()
+
+	client, err := rpc.Dial("http://localhost:8545")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.Close()
+
+	res := make(map[string]interface{})
+	err = client.Call(&res, "rpc_discover")
+	if err != nil {
+		t.Errorf("client call: %v", err)
+	}
+
+	openRPCSpecURL := "https://github.com/open-rpc/meta-schema/releases/download/1.14.0/open-rpc-meta-schema.json"
+
+	schemaLoader := gojsonschema.NewSchemaLoader()
+	schemaLoader.Draft = gojsonschema.Draft7 // force schema version
+	schemaLoader.AutoDetect = false
+	schema, err := schemaLoader.Compile(gojsonschema.NewReferenceLoader(openRPCSpecURL))
```

Comment:
I'd be tempted to use a local copy of this... otherwise the tests depend on an internet connection... :-S

Reply:
(Rel https://github.com/etclabscore/core-geth/pull/295/files#diff-dc4c814863b683df45708c0e8241d15e11bd148209ffd858351686929c111e2bR281)

Reply:
Oh, I thought I made this question on the description, but I probably forgot it. 
Yes I don't like either that we make the tests depend on the internet, though I liked the idea of pointing the URL to the latest spec version.

Let's bring the file locally and loaded from there

Reply:
Maybe we could add a CI-only test that cross-references the remote document with the local one? (The CI can be expected always have internet...)

Reply:
Solved in https://github.com/etclabscore/core-geth/pull/295/commits/9912e4c24875818472045b503ff67bbd6c1cc924

Do you prefer the file placed somewhere else? Or maybe within a **testdata** folder

Reply:
@meowsbits do you mean in the style of a cron GitHub Action or a go test marked for CI only? Do you want to do this now, or open an issue for later?

Reply:
I think we can separate it into it's own issue; IMO now this is doing exactly what it set out to do ✅. 
It might be nice to _be able_ to run the internet-requiring cross-referencing (:sweat_smile:) test locally too... maybe something like

```go
var openRPCSpecLatestTestEnvVarName := "COREGETH_SOME_GOOD_NAME"
func TestOpenRPCMetaSchemaSpecLatest(t *testing.T) {
    if os.Getenv("CI") == "" && os.Getenv(openRPCSpecLatestTestEnvVarName) {
        t.Skip()
    }
}
```

Reply:
actually, I am searching for things to work on, so let me tackle the test tomorrow in this PR. :)